ssl/ssl_local.h

```diff
@@ -1815,6 +1816,13 @@ struct ssl_connection_st {
     SSL_CONNECTION_FROM_SSL_ONLY_int(ssl, const)
 # define SSL_CONNECTION_GET_CTX(sc) ((sc)->ssl.ctx)
 # define SSL_CONNECTION_GET_SSL(sc) (&(sc)->ssl)
+# define SSL_CONNECTION_PROV_QUERRY(spq) \
+      (spq != NULL ? \
+          (SSL_CONNECTION_GET_SSL(spq) && SSL_CONNECTION_GET_SSL(spq)->propq != NULL ? \
+          SSL_CONNECTION_GET_SSL(spq)->propq : SSL_CONNECTION_GET_CTX(spq)->propq) : NULL)
```

Comment:
The indention here is quite off and it's fairly complicated.
Maybe consider making this a function and spreading everything nicely?